protocols/gossipsub/src/behaviour/tests.rs

```diff
@@ -895,4 +895,26 @@ mod tests {
         // Peers should be removed to reach mesh_n
         assert_eq!(gs.mesh.get(&topics[0]).unwrap().len(), config.mesh_n);
     }
+
+    // Some very basic test of public api methods.
+    #[test]
+    fn test_public_api() {
```

Comment:
This test fails with:

```
---- behaviour::tests::tests::test_public_api stdout ----
thread 'behaviour::tests::tests::test_public_api' panicked at 'assertion failed: `(left == right)`
  left: `[PeerId("QmQ8i9Cjq4Ge38wD8giUmVmQ5PASETyVoBTbsitZq6UcDV"), PeerId("QmYqtVQB37rh1HKnonYEAhiWwfxFjFNHAR9LKY3Q2EGw6c"), PeerId("QmYrr9uGZ6AcA14cVfZzoaE7QE1e4A8UxozrgDu4AWWEjh"), PeerId("QmchgnDGecCQw5MBAKoECKdcjSn9JZDdbZ3DmZTCFHSZzH")]`,
 right: `[PeerId("QmYqtVQB37rh1HKnonYEAhiWwfxFjFNHAR9LKY3Q2EGw6c"), PeerId("QmchgnDGecCQw5MBAKoECKdcjSn9JZDdbZ3DmZTCFHSZzH"), PeerId("QmQ8i9Cjq4Ge38wD8giUmVmQ5PASETyVoBTbsitZq6UcDV"), PeerId("QmYrr9uGZ6AcA14cVfZzoaE7QE1e4A8UxozrgDu4AWWEjh")]`: Expected peers for a registered topic to contain all peers.', protocols/gossipsub/src/behaviour/tests.rs:910:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I would guess you want to sort the vectors or peer ids before comparing them.

Reply:
Yeah. Had a branch where I am using order based collections so this is not an issue. Pushed a fix.